src/cluster.h

```diff
@@ -12,6 +12,12 @@
 #define CLUSTER_FAIL 1                                              /* The cluster can't work */
 #define CLUSTER_NAMELEN 40                                          /* sha1 hex length */
 
+/* Reason why the cluster state changes to fail. When adding new reasons,
+ * make sure to update clusterLogFailReason. */
+#define CLUSTER_FAIL_NONE 0
+#define CLUSTER_FAIL_NOT_FULL_COVERAGE 1
+#define CLUSTER_FAIL_MINORITY_PARTITION 2
```

Comment:
If we use `enum` and don't add a `default:` case in the switch, there will be a compile warning for the switch if we add another value and forget to add a case for it in the switch. Just an idea. (I learned it from Madelyn.)

OTOH, the style in this file is to use integer and define, so we can follow this style, as you prefer.